Add the doc on Gateway (#2730)
codenvy/che-docs: modules/end-user-guide/pages/idea-ultimate.adoc

[id="idea-ultimate"]

> = JetBrains IntelliJ IDEA Ultimate Edition (over JetBrains Gateway)

Reviewer: ```suggestion
= Connecting JetBrains IntelliJ IDEA Ultimate Edition to a {prod-short} workspace
```

> Eclipse Che supports connecting your local IntelliJ IDE to a running Che workspace.

Reviewer: ```suggestion
{prod} supports connecting your local IntelliJ IDE to a running {prod-short} workspace over JetBrains Gateway.
```

.Prerequisites

> . Install the JetBrains Gateway application from https://www.jetbrains.com/remote-development/gateway/

Reviewer: ```suggestion
. link:https://www.jetbrains.com/remote-development/gateway/[The JetBrains Gateway application] is installed.
```

> . Install the Gateway provider for OpenShift Dev Spaces from https://plugins.jetbrains.com/plugin/24234-openshift-dev-spaces

Reviewer: ```suggestion
. link:https://plugins.jetbrains.com/plugin/24234-openshift-dev-spaces[The Gateway provider for OpenShift Dev Spaces] is installed.
```

> . In your local terminal, log in to your OpenShift server with the OpenShift Client. `oc login` command establishes the authenticated session and saves the connection information to the configuration file which is read by the Gateway provider for OpenShift Dev Spaces.

Reviewer: ```suggestion
. In your local terminal, you are logged in to your OpenShift server with the OpenShift Client.
+
[NOTE]
====
The `oc login` command establishes the authenticated session and saves the connection information to the configuration file which is read by the Gateway provider for OpenShift Dev Spaces.
====

> .Procedure of connecting to a new workspace

Reviewer: ```suggestion
.Connecting JetBrains IntelliJ IDEA Ultimate Edition to a new workspace

.Procedure
```

> . When creating your workspace on Che Dashboard, choose `IntelliJ IDEA Ultimate (desktop)` editor:

Reviewer: ```suggestion
. Create a workspace on the {prod-short} Dashboard and choose `IntelliJ IDEA Ultimate (desktop)` editor:
```

+
image::editor-idea-iu.png[IntelliJ IDEA Ultimate on Dashboard,link="{imagesdir}/editor-idea-iu.png"]

> . Once the workspace runs, the Dashboard will ask if you want to open your local JetBrains Gateway application:

Reviewer: ```suggestion
. Wait for the prompt to open your local JetBrains Gateway application to appear:
```

+
image::open-gateway-prompt.png[Open Gateway prompt,link="{imagesdir}/open-gateway-prompt.png"]

> Clicking the `Open Gateway` button will start your local JetBrains Client application connected to your Dev Spaces workspace:

Reviewer: ```suggestion
Click the `Open Gateway` button to start your local JetBrains Client application connected to your Dev Spaces workspace:
```

image::gateway-connecting.png[Connecting to remote host,link="{imagesdir}/gateway-connecting.png"]

> .Procedure of connecting to an existing workspace

Reviewer: ```suggestion
.Connecting JetBrains IntelliJ IDEA Ultimate Edition to an existing workspace
```

> There are two options to connect to an already existing workspace:

Reviewer: ```suggestion
There are two options to connect the local IntelliJ IDE to an existing workspace:
```

* from the Che Dashboard
* from the Gateway application

> The simplest way to connect to an existing workspace is to start a workspace from the Che Dashboard and click the `Open Gateway` button.

Reviewer: ```suggestion
The simplest way to connect to an existing workspace is to start a workspace from the {prod-short} Dashboard and click the `Open Gateway` button.
```

> Another option to connect is using the Gateway application. It allows connecting without accessing the Che Dashboard.

Reviewer: ```suggestion
However, with the Gateway application, you can connect the IntelliJ IDE to your workspace without accessing the {prod-short} Dashboard.
```

> . Open the Gateway app and click `Connect to Dev Spaces`:

Reviewer: ```suggestion
.Procedure

. Open the Gateway app and click `Connect to Dev Spaces`:
```

+
image::gateway.png[JetBrains Gateway main window,link="{imagesdir}/gateway.png"]

. On the next page, provide the parameters to connect to the OpenShift API server and click the `Check Connection and Continue` button:
+
image::gateway-connect.png[Connecting to OpenShift API server,link="{imagesdir}/gateway-connect.png"]

. Choose your workspace and click the `Connect` button:
+
image::gateway-select-ws.png[Connecting to OpenShift API server,link="{imagesdir}/gateway-select-ws.png"]

> Your local Gateway application will run the JetBrains Client and connect to the workspace.

Reviewer: ```suggestion

.Verification

. Your local Gateway application is running the JetBrains Client and connects to the workspace.
+
```
